src/actor.rs

```diff
@@ -0,0 +1,94 @@
+use std::marker::PhantomData;
+use crate::traits::{LoadableState, UptimeCheckerActor};
+use crate::types::{InitParams, MultiAddr, NodeInfo, NodeInfoPayload, PeerID, ReportPayload};
+use crate::{ensure, Error};
+
+const THRESHOLD_NUM: usize = 20000;
+const THRESHOLD_DUM: usize = 30000;
+
+pub struct Actor<S: LoadableState> {
+    _phantom: PhantomData<S>
+}
+
+impl <S: LoadableState> UptimeCheckerActor for Actor<S> {
+    fn init(params: InitParams) -> Result<(), Error> {
+        // TODO: perform simple checks
+        let len = params.ids.len();
+        let mut nodes = vec![];
+        for i in 0..len {
+            nodes.push(NodeInfo::new(
+                PeerID::from(&params.ids[i]),
+                params.creators[i],
+                params.addresses[i]
+                    .iter()
+                    .map(MultiAddr::from)
+                    .collect(),
+            ));
+        }
+        let state = S::new(nodes)?;
+        state.save()?;
+        Ok(())
+    }
+
+    fn new_checker(c: NodeInfoPayload) -> Result<(), Error> {
+        let mut s = S::load()?;
+        s.upsert_checker(NodeInfo::from(c))?;
+        s.save()?;
+        Ok(())
+    }
+
+    fn new_member(m: NodeInfoPayload) -> Result<(), Error> {
+        let mut s = S::load()?;
+        s.upsert_node(NodeInfo::from(m))?;
+        s.save()?;
+        Ok(())
+    }
+
+    fn edit_checker(c: NodeInfoPayload) -> Result<(), Error> {
```

Comment:
I don't know if we may be introducing a bug here. When you edit and remove a checker/member, we don't need to check that the owner set in the method arguments are the same (as we do for `new`), but that the owner for the checker/member persisted on-chain is the caller of `edit/rm`. We want to give checkers and members the ability of changing the owner keys of their checkers. 

Let me know if I misunderstood something and this is already the case.

Reply:
![image](https://user-images.githubusercontent.com/108330426/192534938-800ee48e-5d99-453c-a570-11975035ce36.png)
I think currently it is checking the caller and the node stored in the map on chain. It should be aligned with `the owner for the checker/member persisted on-chain is the caller of edit/rm`